components/dashboard/src/components/SelectIDEComponent.tsx

```diff
@@ -0,0 +1,145 @@
+            )) {
+                result.push({
+                    id: ide.id,
+                    element: <IdeOptionElementInDropDown option={ide} useLatest={false} />,
+                    isSelectable: true,
+                });
+                if (ide.imageVersion !== ide.latestImageVersion) {
+                    result.push({
+                        id: ide.id + "-latest",
+                        element: <IdeOptionElementInDropDown option={ide} useLatest={true} />,
+                        isSelectable: true,
+                    });
+                }
+            }
+            return result;
+        };
+    }, [ideOptions]);
+    const internalOnSelectionChange = (id: string) => {
+        const { ide, useLatest } = parseId(id);
+        props.onSelectionChange(ide, useLatest);
+    };
+    const ide = props.selectedIdeOption || ideOptions?.defaultIde || "";
+    return (
+        <DropDown2
+            getElements={getElements}
+            onSelectionChange={internalOnSelectionChange}
+            searchPlaceholder={"Select Editor"}
+        >
+            <IdeOptionElementSelected option={ideOptions?.options[ide]} useLatest={!!props.useLatest} />
+        </DropDown2>
+    );
+}
+
+function parseId(id: string): { ide: string; useLatest: boolean } {
+    const useLatest = id.endsWith("-latest");
+    const ide = useLatest ? id.slice(0, -7) : id;
+    return { ide, useLatest };
+}
+
+interface IdeOptionElementProps {
+    option: IDEOption | undefined;
+    useLatest: boolean;
+}
+
+function IdeOptionElementSelected(p: IdeOptionElementProps): JSX.Element {
+    const { option, useLatest } = p;
+    if (!option) {
+        return <></>;
+    }
+    const version = useLatest ? option.latestImageVersion : option.imageVersion;
+    const label = option.type === "desktop" ? "" : option.type;
+
+    return (
+        <div className="flex" title={option.title}>
+            <div className="mx-2 my-2">
+                <img className="w-8 filter-grayscale self-center" src={Editor} alt="logo" />
+            </div>
+            <div className="flex-col ml-1 mt-1 flex-grow">
+                <div className="text-gray-700 font-semibold">Editor</div>
+                <div className="flex text-xs text-gray-500 dark:text-gray-600">
+                    <div className="font-semibold">{option.title}</div>
+                    {version && (
+                        <>
+                            <div className="mx-1">&bull;</div>
+                            <div>{version}</div>
+                        </>
+                    )}
+                    {label && (
+                        <>
+                            <div className="mx-1">&bull;</div>
+                            <div>{label && label[0].toLocaleUpperCase() + label.slice(1)}</div>
+                        </>
+                    )}
+                    {useLatest && <div className="ml-2 rounded-xl bg-gray-200 px-2">Latest</div>}
```

Comment:
**suggestion**: Ideally, this would use the `PillLabel` component variant being added in https://github.com/gitpod-io/gitpod/pull/15397, but we can do that later, see relevant comment in https://github.com/gitpod-io/gitpod/pull/15397#discussion_r1052284501.

Reply:
I tried, but it's too big.